Generate diffs for helm values.yaml file between releases (#3420)
istio/istio.io: content/docs/reference/config/installation-options-version-comparison/index.md

---

> title: Installation Options - Generate differences between current release and previous release (v1.0)

Reviewer: Can you shorten the title?  Something like "Installation options added and removed in 1.1"

Author: @sdake I have given it a try. Will let you verify.  Thanks.

Reviewer 2: I think this document should be hidden and not even have a title in the index.
There should be a link to it from the `Installation Options` page, right at the top. Something like:
```
See [installation option release changes ](<link>) for the differences between the current release and the previous one.
```

@geeknoid @rcaballeromx WDYT?

Reviewer 3: Why hide it? If it's important enough to be on the web site, then shouldn't be listed in the normal index?

The site doesn't currently have support for content that doesn't show up in the various indices. I'd have to add some annotations to induce this behavior.

Reviewer 2: Hidden or not, I think the most important thing is that there is a link to it from the `Installation Options` page, which IMO is the reference document that people would go to for options info.

The release-specific changes, if not hidden, isn't really a reference doc to me, that's why I think it seems out of place. Maybe there's a good place for something like this somewhere closer to the release notes?

Reviewer: @frankbu where we put it I leave up to you and @geeknoid.  I do believe this is helpful for operators to get a feel for what has changed.  I'm not sure this PR will merge prior to friday, but I'd like this work to backport to 1.1 when it is ready.

Cheers
-steve

description: Describes the differences between the current and previous releases of installation options from the Helm charts.
weight: 30
keywords: [kubernetes,helm]
---

> 

Reviewer: need some type of introduction here.  The reader will see a table and have no idea what they are looking at :)

Author: done

## `ingress` options

| Old Key | New Key | Old Default Value | New Default Value | Old Description | New Description|
| --- | --- | --- | --- | --- | --- |

> | `ingress.service.ports.nodePort` | `n/a` | `32000` | `n/a` | `` | `n/a` |

Reviewer: This is significantly more readable.  I wonder as a future followup if it would be possible to classify keys that were added, vs keys that were removed.  A deletion (`n/a/`) in New Key could be one table, while an addition (`n/a` in old key) could be ea different table.

Author: @sdake I have tried to address this in the latest PR. Not sure if this is what you had in mind. Let me know if we need to make any changes. Thanks.

Reviewer: @johnma14 I think you are headed in the right direction.  I have added some comments and asked for some information architecture expertise from @rcaballeromx.  Note @rcaballeromx this document is autogenerated from yaml files, so coming up with a good structure to begin with keeps the churn on the generation implementation to a minimum.

---

> title: Installation Options - Generate differences between release-1.0 and release-1.1

Reviewer 4: The title is misleading. The reader won't generate the tables.

```suggestion
title: Helm Installation Options Differences Between Release-1.0 and Release-1.1
```

> description: Describes the differences between the current and previous releases of installation options from the Helm charts.

Reviewer 4: ```suggestion
description: Details the Helm chart installation options differences between release-1.0 and release-1.1.
```

weight: 30

> keywords: [kubernetes,helm]

Reviewer 4: ```suggestion
keywords: [kubernetes,helm, install, options]
```

---

> The table below shows the comparison of the installation options (used to customize Istio install using Helm) between release 1.0 and release 1.1. These include new installation options that have been added in the new release, installation options that were present in the previous release but removed from the new release and installation options whose values or descriptions have been modified in the current release.

Reviewer 4: ```suggestion
The tables below show changes made to the installation options used to customize Istio install using Helm between release 1.0 and release 1.1. The tables are grouped in four different categories:

- The existing installation options that remain unchanged.
- The new installation options added in the new release.
- The installation options already in the previous release but whose values or descriptions have been modified in the 
  new release.
- The installation options removed from the new release.
```

<!-- Run python scripts/tablegen.py to generate this table -->

<!-- AUTO-GENERATED-START -->

> 

Reviewer: can you add a section `## Unmodified configuration options`

Reviewer 4: Begin with the known. The existing options that were not changed.

> # Comparison of configuration options between release 1.0 and current release 

Reviewer 4: Continue with the slightly unfamiliar. After covering what hasn't changed, you cover what was changed but slightly. I think that works well but keep the heading short, specific, yet descriptive.

```suggestion
## Changed configuration options 
```

> ## `tracing` options

Reviewer 4: The heading should not be `tracing` options. To make it easy for readers to know where in the doc they are, the headings for each option should provide that signal. Tracing is the option, what is listed in the table are the modified key/value pairs. Making that explicit on each option heading will help your readers know instantly if the information listed is relevant to them.

```suggestion
###  Changed `tracing` key/value pairs
```

| `gateways.istio-ilbgateway.ports.name` | `tcp-dns` | `grpc-pilot-mtls` | `` | `` |
| `gateways.istio-ilbgateway.ports.name` | `tcp-dns` | `grpc-pilot` | `` | `` |
| `gateways.istio-ilbgateway.ports.name` | `tcp-dns` | `tcp-citadel-grpc-tls` | `` | `` |
| `gateways.istio-ilbgateway.secretVolumes.secretName` | `istio-ilbgateway-ca-certs` | `istio-ilbgateway-certs` | `` | `` |
| `gateways.istio-ilbgateway.secretVolumes.mountPath` | `/etc/istio/ilbgateway-ca-certs` | `/etc/istio/ilbgateway-certs` | `` | `` |

## `certmanager` options

| Key | Old Default Value | New Default Value | Old Description | New Description|
| --- | --- | --- | --- | --- |
| `certmanager.tag` | `v0.3.1` | `v0.6.2` | `` | `` |

## `kiali` options

| Key | Old Default Value | New Default Value | Old Description | New Description|
| --- | --- | --- | --- | --- |
| `kiali.tag` | `istio-release-1.0` | `v0.14` | `` | `` |

## `pilot` options

| Key | Old Default Value | New Default Value | Old Description | New Description|
| --- | --- | --- | --- | --- |
| `pilot.autoscaleMax` | `1` | `5` | `` | `` |
| `pilot.traceSampling` | `100.0` | `1.0` | `` | `` |

> # New configuration options that have been added to the current release 

Reviewer 4: Next, you cover what as added, the new. I think that is good.
```suggestion
## New configuration options
```

> ## `istio_cni` options

Reviewer 4: To make it easy for readers to know on what part of the document they are, each of this headings should provide that signal. In this section, all headings should indicate that the key/value pairs are new to the option in question.

```suggestion
## New `istio_cni` key/value pairs
```

> | New Key | New Default Value | New Description|

Reviewer 4: The label "New" on the table headings is not necessary here, since there is no old. 
```suggestion
| Key | Default Value | Description |
```

| `istiocoredns.replicaCount` | `1` | `` |
| `istiocoredns.coreDNSImage` | `coredns/coredns:1.1.2` | `` |
| `istiocoredns.coreDNSPluginImage` | `istio/coredns-plugin:0.1-istio-1.1` | `` |
| `istiocoredns.nodeSelector` | `{}` | `` |

## `security` options

| New Key | New Default Value | New Description|
| --- | --- | --- |
| `security.enabled` | `true` | `` |
| `security.createMeshPolicy` | `true` | `` |
| `security.nodeSelector` | `{}` | `` |

## `nodeagent` options

| New Key | New Default Value | New Description|
| --- | --- | --- |
| `nodeagent.enabled` | `false` | `` |
| `nodeagent.image` | `node-agent-k8s` | `` |
| `nodeagent.env.CA_PROVIDER` | `""` | `` |
| `nodeagent.env.CA_ADDR` | `""` | `` |
| `nodeagent.env.Plugins` | `""` | `` |
| `nodeagent.nodeSelector` | `{}` | `` |

## `pilot` options

| New Key | New Default Value | New Description|
| --- | --- | --- |
| `pilot.autoscaleEnabled` | `true` | `` |
| `pilot.env.PILOT_PUSH_THROTTLE_COUNT` | `100` | `` |
| `pilot.env.GODEBUG` | `gctrace=1` | `` |
| `pilot.cpu.targetAverageUtilization` | `80` | `` |
| `pilot.nodeSelector` | `{}` | `` |
| `pilot.keepaliveMaxServerConnectionAge` | `30m` | `` |

> # Old configuration options that have been removed from the current release

Reviewer: `## Removed configuration options`

Reviewer 4: It's good that you end with the removed options. That way you convey a sense of progress. I agree with @sdake on the language of the heading:
```suggestion
## Removed configuration options
```

> ## `ingress` options

Reviewer: these sections should be `###`

> ## `ingress` options

Reviewer 4: The headings for each option in this section should let readers know that the key/value pairs are no longer available to them. 
```suggestion
### Removed `ingress` key/value pairs
```

> | Old Key | Old Default Value | Old Description |

Reviewer 4: The adjective "old" is irrelevant to readers if it is clear from the heading that the table contains the removed key/value pairs as per my previous suggestion.

```suggestion
| Key | Default Value | Description |
```
